src/linagora.esn.calendar/app/calendar/calendar-view/calendar-view.controller.spec.js

```diff
@@ -429,7 +433,9 @@ describe('The calendarViewController', function() {
       expect(this.scope.calendars).to.shallowDeepEqual([this.calendars[0], { uniqueId: updatedCalendar.uniqueId, color: updatedCalendar.color }]);
       expect(this.scope.eventSourcesMap[updatedCalendar.getUniqueId()].backgroundColor).to.equal(updatedCalendar.color);
       expect(fullCalendarSpy).to.have.been.calledWith('removeEventSource', sinon.match.has('uniqueId', updatedCalendar.uniqueId));
-      expect(fullCalendarSpy).to.have.been.calledWith('addEventSource', sinon.match.has('uniqueId', updatedCalendar.uniqueId));
+      setTimeout(function() {
+        expect(fullCalendarSpy).to.have.been.calledWith('addEventSource', sinon.match.has('uniqueId', updatedCalendar.uniqueId));
+      }, 100);
```

Comment:
Why use `setTimeout` here? You can use `$rootScope.$digest()` in the tests that have promises.

Reply:
without a timeout fullCalendarSpy is always called with removeEventSource , i will try your solution